include/nbl/asset/IDescriptorSet.h

```diff
@@ -93,34 +95,32 @@ class IDescriptorSet : public virtual core::IReferenceCounted // TODO: try to re
 				}
 				~SDescriptorInfo()
 				{
-					if (desc and (desc->getTypeCategory()==IDescriptor::EC_IMAGE or desc->getTypeCategory() == IDescriptor::EC_SAMPLER))
-						info.image.sampler = nullptr;
```

Comment:
you just broke my code, look at what the original did

Reply:
Since the `SDescriptorInfo` owns the `SBufferImageInfo info` struct I thought this would happen regardless on destruction. Rolling it back

Reply:
no because its in a union, so you loose any implicit construction/destruction of a member!